grafanacloud_key.go

```
@@ -56,13 +92,23 @@ func (b *grafanaCloudBackend) keyRevoke(ctx context.Context, req *logical.Reques
 	}
 
 	org := config.Organisation
-	tokenID := req.Secret.InternalData["name"].(string)
-	err = c.DeleteCloudAPIKey(org, tokenID)
+	tokenName := req.Secret.InternalData["name"].(string)
+	apiType, found := req.Secret.InternalData["type"].(string)
+
+	if !found || apiType == "Cloud" {
+		err = c.DeleteCloudAPIKey(org, tokenName)
+	} else {
+		tokenID := req.Secret.InternalData["id"].(string)
+		stackSlug := req.Secret.InternalData["stack_slug"].(string)
+		id, _ := strconv.ParseInt(tokenID, 10, 64)
```

if we store the id as an int can we avoid this parsing?

---

Same as above

---

Ok, I see in that case I think we should either handle the parsing error here or treat it as a float64 and cast it back to an int but I don't think we should ignore the error.